packages/app-contracts/src/Contracts/index.tsx

```diff
@@ -0,0 +1,134 @@
+// Copyright 2017-2019 @polkadot/app-staking authors & contributors
+// This software may be modified and distributed under the terms
+// of the Apache-2.0 license. See the LICENSE file for details.
+
+import { I18nProps } from '@polkadot/ui-app/types';
+import { ComponentProps } from '../types';
+
+import React from 'react';
+import { RouteComponentProps } from 'react-router';
+import { withRouter } from 'react-router-dom';
+import { Button, CardGrid, Icon } from '@polkadot/ui-app';
+import { withMulti } from '@polkadot/ui-api';
+
+import translate from '../translate';
+import Add from './Add';
+import Contract from './Contract';
+import Call from './Call';
+
+type Props = ComponentProps & I18nProps & RouteComponentProps;
+
+type State = {
+  isAddOpen: boolean,
+  isCallOpen: boolean,
+  callAddress: string | null,
+  callMethod: string | null
+};
+
+class Contracts extends React.PureComponent<Props, State> {
+  state: State = {
+    callAddress: null,
+    callMethod: null,
+    isAddOpen: false,
+    isCallOpen: false
+  };
+
+  render () {
+    const { accounts, basePath, contracts, hasCode, showDeploy, t } = this.props;
+    const { callAddress, callMethod, isAddOpen, isCallOpen } = this.state;
+
+    return (
+      <>
+        <CardGrid
+          emptyText={t('No contracts available')}
+          buttons={
+            <Button.Group>
+              {hasCode && (
+                <>
+                  <Button
+                    isPrimary
+                    label={
+                      <>
+                        <Icon name='cloud upload' />
+                        {'  '}
+                        {t('Deploy a code hash')}
+                      </>
+                    }
+                    onClick={showDeploy()}
+                  />
+                  <Button.Or />
+                </>
+              )}
+              <Button
+                label={
+                  <>
+                    <Icon name='add' />
+                    {'  '}
+                    {t('Add an existing contract')}
+                  </>
+                }
+                onClick={this.showAdd}
+              />
+            </Button.Group>
+          }
+        >
+          {accounts && contracts && Object.keys(contracts).map((address) => {
```

Comment:
Not 100% sure why this is `accounts && contracts &&`? Surely just `contracts && ` should do the trick, cannot see accounts used.